backend/src/routes/user.ts

```diff
@@ -77,6 +76,58 @@ r.get('/:userid', (request, response) => {
   return response.json(userProfile);
 });
 
+/**
+ * -------------------------------------
+ * Route
+ * -------------------------------------
+ */
+r.delete('/:userid', (request, response) => {
+  console.log(`Delete user with ID: ${request.params.userid}`);
+  userService.deleteUser(request.params.userid);
+  return response.json(`Deleted user with ID: ${request.params.userid}`);
+});
+
+/**
+ * -------------------------------------
+ * Route
+ * -------------------------------------
+ */
+ r.patch('/:userid', (request, response) => {
+  const patch = request.body;
+  const userId = request.params.userid;
+  console.log(`User ID: ${userId}`)
+  console.log(request.body)
+
+  // Get current user profile
+  const existingUser : IUserProfile = userService.getUserProfileById(userId);
+  if (existingUser === undefined) {
+    return response.json(`No user found for ID ${userId}`);
+  }
+
+  // Create copy of current user to update
+  const userProfile : IUserProfile = {...existingUser};
+
+  // Update current user profile if patch contains entry
+  userProfile.name = patch.name ? patch.name : userProfile.name;
+  if (patch.name)
+  {
+    userProfile.initials = patch.name.split(" ").length > 1 ?
+      patch.name.split(" ")[0][0].toUpperCase()+patch.name.split(" ")[1][0].toUpperCase()
+      :
+      patch.name[0].toUpperCase();
+  }
+  userProfile.gender = patch.gender ? patch.gender : userProfile.gender;
```

Comment:
A more compact way of writing this would be: 
```js
userProfile.gender = patch.gender || userProfile.gender
```
Can be applied in multiple places in this file

Reply:
eh... javascript is weird... :S